runtime/runtime-standalone/src/lib.rs

```diff
@@ -0,0 +1,414 @@
+use near_pool::{types::PoolIterator, TransactionPool};
+use near_primitives::{
+    account::Account,
+    errors::RuntimeError,
+    hash::CryptoHash,
+    receipt::Receipt,
+    state_record::StateRecord,
+    transaction::{ExecutionOutcome, ExecutionStatus, SignedTransaction},
+    types::AccountId,
+    types::{Balance, BlockHeight, EpochHeight, Gas, StateChangeCause},
+};
+use near_runtime_configs::RuntimeConfig;
+use near_store::{get_account, set_account, Store, Trie, TrieUpdate};
+use node_runtime::{state_viewer::TrieViewer, ApplyState, Runtime};
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+const DEFAULT_GENESIS_HEIGHT: u64 = 3;
+
+#[derive(Debug)]
+pub struct GenesisConfig {
+    pub genesis_time: u64,
+    pub gas_price: Balance,
+    pub gas_limit: Gas,
+    pub genesis_height: u64,
+    pub epoch_length: u64,
+    pub runtime_config: RuntimeConfig,
+    pub state_records: Vec<StateRecord>,
+}
+
+impl Default for GenesisConfig {
+    fn default() -> Self {
+        Self {
+            genesis_time: 0,
+            gas_price: 0,
+            gas_limit: std::u64::MAX,
+            genesis_height: 0,
+            epoch_length: DEFAULT_GENESIS_HEIGHT,
+            runtime_config: RuntimeConfig::default(),
+            state_records: vec![],
+        }
+    }
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct Block {
+    prev_block: Option<Box<Block>>,
+    state_root: CryptoHash,
+    gas_burnt: Gas,
+    pub epoch_height: EpochHeight,
+    pub block_height: BlockHeight,
+    pub block_timestamp: u64,
+    pub gas_price: Balance,
+    pub gas_limit: Gas,
+}
+
+impl Block {
+    pub fn genesis(genesis_config: &GenesisConfig) -> Self {
+        Self {
+            prev_block: None,
+            state_root: CryptoHash::default(),
+            block_height: genesis_config.genesis_height,
+            epoch_height: 0,
+            block_timestamp: genesis_config.genesis_time,
+            gas_price: genesis_config.gas_price,
+            gas_limit: genesis_config.gas_limit,
+            gas_burnt: 0,
+        }
+    }
+
+    pub fn produce(&self, new_state_root: CryptoHash, epoch_length: u64) -> Block {
+        Self {
+            gas_price: self.gas_price,
+            gas_limit: self.gas_limit,
+            block_timestamp: self.block_timestamp + 1,
+            prev_block: Some(Box::new(self.clone())),
+            state_root: new_state_root,
+            block_height: self.block_height + 1,
+            epoch_height: self.block_timestamp + 1 / epoch_length,
```

Comment:
It has to be `(self.block_timestamp + 1) / epoch_length`

Reply:
I would even say it has to be `(self.block_height + 1) / epoch_length`

Reply:
I don't understand what it means to divide `block_timestamp` by `epoch_length`. It makes no sense to me whatsoever.

Reply:
I meant block_height of course 😬. Fixed.